src/features/device-heartbeat/index.ts

```diff
@@ -298,30 +298,32 @@ export class DeviceOnlineStateManager extends EventEmitter<{
 					};
 
 					// raise and event for the state change...
-					switch (nextState) {
-						case DeviceOnlineStates.Timeout:
-							await this.updateDeviceModel(
-								deviceId,
-								DeviceOnlineStates.Timeout,
-							);
-							this.scheduleChangeOfStateForDevice(
-								deviceId,
-								await this.getDeviceOnlineState(deviceId),
-								DeviceOnlineStates.Timeout,
-								DeviceOnlineStates.Offline,
-								API_HEARTBEAT_STATE_TIMEOUT_SECONDS, // put the device into a timeout state if it misses it's scheduled heartbeat window... then mark as offline
-							);
-							break;
-						case DeviceOnlineStates.Offline:
-							await this.updateDeviceModel(
-								deviceId,
-								DeviceOnlineStates.Offline,
-							);
-							break;
-						default:
-							throw new Error(
-								`An unexpected value was encountered for the target device state: ${nextState}`,
-							);
+					if (deviceId) {
+						switch (nextState) {
+							case DeviceOnlineStates.Timeout:
+								await this.updateDeviceModel(
+									deviceId,
+									DeviceOnlineStates.Timeout,
+								);
+								this.scheduleChangeOfStateForDevice(
+									deviceId,
+									await this.getDeviceOnlineState(deviceId),
+									DeviceOnlineStates.Timeout,
+									DeviceOnlineStates.Offline,
+									API_HEARTBEAT_STATE_TIMEOUT_SECONDS, // put the device into a timeout state if it misses it's scheduled heartbeat window... then mark as offline
+								);
+								break;
+							case DeviceOnlineStates.Offline:
+								await this.updateDeviceModel(
+									deviceId,
+									DeviceOnlineStates.Offline,
+								);
+								break;
+							default:
+								throw new Error(
+									`An unexpected value was encountered for the target device state: ${nextState}`,
+								);
+						}
 					}
```

Comment:
It's probably worth it to also raise such cases with something like:
```suggestion
					} else {
						const msg = `Unable to consume API heartbeat state message without a deviceId: ${message}`;
						errors.captureException(new Error(msg), msg);
					}
```

Reply:
Unfortunately I made this PR using the web editor and it's not letting me amend the commit to include the change-type annotation - so if you guys think it's worthwhile to include probably easiest to just recreate.  Either way the issue is solved on my end after running with this change, but might be helpful for others in case they run into this issue.  One alternative approach would be just running a cleanup on the redis queues on startup, similar to how the database migrations occur.

Reply:
Hi @dcaputo-harmoni ,
Let me guide you on how to do this.
1a. In case you do not have a local copy of open-balela-api you will first need to fetch your fork and checkout the branch of this PR with:
```
git clone https://github.com/dcaputo-harmoni/open-balena-api.git
cd open-balena-api
git checkout patch-1
```
1b. In case you already have a local copy of the repo, then you just need to `cd` to it and make sure your local branch is in sync with:
```
git checkout patch-1
git fetch
git reset patch-1 origin:patch-1
```

2. Then you can amend the commit and push with:
```
git commit --amend -m "Skip device heartbeat RSMQ records missing a deviceId property" -m "Change-type: patch"
git push origin patch-1
```

Reply:
Just followed these steps (I needed to merge before the push) but otherwise seemed to work - let me know if it's good to go, and thanks for the tips!